catalog/forms.py

```diff
@@ -0,0 +1,50 @@
+import datetime
+
+from django import forms
+from django.core.exceptions import ValidationError
+from django.utils.translation import ugettext_lazy as _
+
+
+class RenewBookForm(forms.Form):
+    renewal_date = forms.DateField(help_text="Enter a date between now and 4 weeks (default 3).")
+
+    def clean_renewal_date(self):
+        data = self.cleaned_data['renewal_date']
```

Comment:
.cleaned_data có tác dụng gì em ?

Reply:
Tác dụng của cleaned_data là làm sạch và loại bỏ những dữ liệu đầu vào có khả năng không an toàn